asr1k_neutron_l3/models/neutron/l3/firewall.py

```diff
@@ -0,0 +1,264 @@
+        return cls.get_id_by_router_id(rid)
+
+    def __init__(self, router_id: str):
+        self.router_id = router_id
+        self.id = self.get_id_by_router_id(self.router_id)
+        super().__init__()
+
+
+class Zone(FirewallZoneObject):
+
+    PREFIX = const.FWAAS_ZONE_PREFIX
+
+    @property
+    def _rest_definition(self):
+        return ncZone(id=self.id)
+
+
+class DanglingZone(Zone):
+
+    def update(self):
+        return self.delete()
+
+
+class ZonePair(FirewallZoneObject):
+
+    PREFIX = const.FWAAS_ZONE_PAIR_PREFIX
+    DEFAULT_ALLOW_INSPECT_POLICY = const.FWAAS_DEFAULT_ALLOW_INSPECT_POLICY
+
+    def __init__(self, router_id: str, source: str, destination: str, policy_id: Optional[str]):
+        super().__init__(router_id)
+        self.source = source
+        self.destination = destination
+        self.policy_id = policy_id
+        if self.policy_id is not None:
+            self.service_policy = ServicePolicy.get_id_by_policy_id(self.policy_id)
+        else:
+            self.service_policy = self.DEFAULT_ALLOW_INSPECT_POLICY
+
+    @property
+    def _rest_definition(self):
+        return ncZonePair(id=self.id, source=self.source, destination=self.destination,
+                        service_policy=self.service_policy)
+
+
+class ZonePairExtEgress(ZonePair):
+
+    PREFIX = const.FWAAS_ZONE_PAIR_EXT_EGRESS_PREFIX
+
+    def __init__(self, router_id: str, policy_id: Optional[str] = None):
+        self.source = 'default'
+        self.destination = Zone.get_id_by_router_id(router_id)
+        super().__init__(router_id, self.source, self.destination, policy_id)
+
+
+class DanglingZonePairExtEgress(ZonePairExtEgress):
+
+    def __init__(self, router_id: str):
+        super().__init__(router_id)
+        self.service_policy = None
+
+    def update(self):
+        return self.delete()
+
+
+class ZonePairExtIngress(ZonePair):
+
+    PREFIX = const.FWAAS_ZONE_PAIR_EXT_INGRESS_PREFIX
+
+    def __init__(self, router_id: str, policy_id: Optional[str] = None):
+        self.source = Zone.get_id_by_router_id(router_id)
+        self.destination = 'default'
+        super().__init__(router_id, self.source, self.destination, policy_id)
+
+
+class DanglingZonePairExtIngress(ZonePairExtIngress):
+
+    def __init__(self, router_id: str):
+        super().__init__(router_id)
+        self.service_policy = None
+
+    def update(self):
+        return self.delete()
+
+
+class FirewallVrfPolicer(base.Base):
+
+    DEFAULT_PARAMETER_MAP = const.FWAAS_DEFAULT_PARAMETER_MAP
+
+    def __init__(self, router_id: str, parameter_map=None) -> None:
+        if parameter_map is None:
+            parameter_map = self.DEFAULT_PARAMETER_MAP
+        self.parameter_map = parameter_map
+        self.router_id = router_id
+
+    @property
+    def vrf(self) -> str:
+        return utils.uuid_to_vrf_id(self.router_id)
+
+    @property
+    def id(self) -> str:
+        return self.vrf
```

Comment:
I think I took this over because `NyBase` does a lot of back and forth mapping between `neutron_router_id` and `id` but I believe in this higher level class it is not needed, at least I couldn't grep any usage of this.

Reply:
Link not working anymore, but I'm going to trust you on this :)